lib/seattleflu/id3c/cli/command/etl/redcap_det_scan.py

```diff
@@ -81,35 +81,48 @@ def redcap_det_scan(*, db: DatabaseSession, cache: TTLCache, det: dict, redcap_r
         LOG.warning("Skipping enrollment with insufficient information to construct patient")
         return None
 
-    encounter_entry, encounter_reference = create_encounter(redcap_record, patient_reference, site_reference, location_resource_entries)
+    primary_encounter_entry, primary_encounter_reference = create_primary_encounter(
+        redcap_record, patient_reference,
+        site_reference, location_resource_entries)
 
-    if not encounter_entry:
-        LOG.warning("Skipping enrollment with insufficient information to construct an encounter")
+    if not primary_encounter_entry:
+        LOG.warning("Skipping enrollment with insufficient information to construct a primary encounter")
         return None
 
-    questionnaire_entry = create_questionnaire_response(redcap_record, patient_reference, encounter_reference)
+    questionnaire_entry = create_questionnaire_response(
+        redcap_record, patient_reference, primary_encounter_reference)
 
     specimen_entry = None
     specimen_observation_entry = None
     specimen_received = is_complete('post_collection_data_entry_qc', redcap_record)
 
     if specimen_received:
         specimen_entry, specimen_reference = create_specimen(redcap_record, patient_reference)
-        specimen_observation_entry = create_specimen_observation_entry(specimen_reference, patient_reference, encounter_reference)
+        specimen_observation_entry = create_specimen_observation_entry(
+            specimen_reference, patient_reference, primary_encounter_reference)
     else:
         LOG.info("Creating encounter for record without sample")
 
     if specimen_received and not specimen_entry:
         LOG.warning("Skipping enrollment with insufficent information to construct a specimen")
         return None
 
+    follow_up_encounter_entry = None
+    follow_up_complete = is_complete('day_7_follow_up', redcap_record)
+
+    if follow_up_complete:
+        # Follow-up encounter for 7 day follow-up survey
```

Comment:
It doesn't look like `follow_up_complete` is used again, so you could just check for `if is_complete('day_7_follow_up', redcap_record)` without creating a new variable.